pkg/registry/core/persistentvolumeclaim/strategy.go

```diff
@@ -66,8 +66,8 @@ func (persistentvolumeclaimStrategy) GetResetFields() map[fieldpath.APIVersion]*
 func (persistentvolumeclaimStrategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
 	pvc := obj.(*api.PersistentVolumeClaim)
 	pvc.Status = api.PersistentVolumeClaimStatus{}
-
-	pvcutil.DropDisabledFields(&pvc.Spec, nil)
+	pvcutil.SetAllocatedResources(pvc, nil)
```

Comment:
TODO for me to see if we have precedent on setting status during create.

Reply:
actually I think there is a precedence for this in `PrepareForCreate` function of podStrategy which sets:

```golang
	pod := obj.(*api.Pod)
	pod.Status = api.PodStatus{
		Phase:    api.PodPending,
		QOSClass: qos.GetPodQOS(pod),
	}

	podutil.DropDisabledPodFields(pod, nil)
```

Reply:
Is the feature able to function properly without setting the status on api create/update? If so, it's preferable not to set it because we're going to have to handle this during an upgrade scenario where existing pvs don't have the status set.

Reply:
yes, the feature works without having the field set. I am going to drop call to `SetAllocatedResources` entirely from this code path.

Reply:
removed